dev/TEAL.md

```diff
@@ -461,6 +462,7 @@ Some of these have immediate data in the byte or bytes after the opcode.
 | 59 | NumLogs | uint64 | v5  | Number of Logs (only with `itxn` in v5). Application mode only |
 | 60 | CreatedAssetID | uint64 | v5  | Asset ID allocated by the creation of an ASA (only with `itxn` in v5). Application mode only |
 | 61 | CreatedApplicationID | uint64 | v5  | ApplicationID allocated by the creation of an application (only with `itxn` in v5). Application mode only |
+| 62 | LastLog | []byte | v6  | The last message emitted. Empty bytes if none were emitted. Application mode only |
```

Comment:
Is it equivalent to the last item of `Logs`?

Looking back at this two comments:
* is Logs/LastLog of the current transaction updated as `log` are performed?
* this list of transaction field does not separate between arrays of `[]byte/uint64` and just `[]byte/uint64`. For eample, `Logs` type should be `[][]byte`. And `Assets` type should be `[]uint64`.

Reply:
Yes, it is the same as the last element of Logs.  It's a convenience because the API places the return value in the last log.  To get it would otherwise require: `itxn NumLogs; int 1; -; itxnas Logs`. Something like 7 bytes and cost 4. Instead of 2 bytes and cost 1 for this.  Adding extra fields to `txn` or `global` doesn't take "opcode space" so since I got this request from two people trying to use the ABI, I thought it was worthwhile.

No, it is not legal to use `txn Logs` or `gtxn n Logs` for an n >= your group index. (Same applies to any of the new "effects" txn fields - things that aren't actually in the original txn, but result from it.)

That is true about types, we have always documented the actual type that gets returned when you use the field: You _must_ access a specific element, there's no such thing as asking for the array.  I don't _love_ it, but don't see a ton of value in add the `[]` either.

Reply:
I think marking it as `[][]byte` and then explaining that you cannot access the full field but use `txna` to access a specific item would simplify reading the specs/doc. Otherwise I think it's a bit confusing. What do you think? (for the future)